extensions/amp-video/1.0/video-wrapper.js

```diff
@@ -0,0 +1,207 @@
+        muted={muted}
+        controls={controls && (!autoplay || userInteracted)}
+        onLoad={() => readyDeferredRef.current.resolve()}
+        onLoadedMetadata={() => {
+          if (!mediasession) {
+            return;
+          }
+
+          const metadata = playerRef.current.getMetadata() || {
+            ...EMPTY_METADATA,
+            artwork: undefined,
+          };
+
+          metadata.title =
+            metadata.title ||
+            passthroughProps.title ||
+            passthroughProps['aria-label'] ||
+            document.title;
+
+          metadata.artwork =
+            metadata.artwork ||
+            parseSchemaImage(document) ||
+            parseOgImage(document) ||
+            parseFavicon(document);
+
+          setMetadata(metadata);
+        }}
+        onPlaying={() => setPlaying(true)}
+        onPause={() => setPlaying(false)}
+        {...passthroughProps}
+      >
+        {children}
+      </Component>
+      {autoplay && !userInteracted && (
+        <Autoplay
+          playing={playing}
+          displayIcon={!noaudio && muted}
+          wrapperRef={wrapperRef}
+          play={play}
+          pause={pause}
+          displayOverlay={controls}
+          onOverlayClick={() => {
+            setMuted(false);
+            setUserInteracted(true);
+          }}
+        />
+      )}
+    </div>
+  );
+}
+
+/**
+ * @param {!VideoAutoplayProps} props
+ * @return {PreactDef.Renderable}
+ */
+function Autoplay({
+  displayIcon,
+  playing,
+  displayOverlay,
+  onOverlayClick,
+  wrapperRef,
+  play,
+  pause,
+}) {
+  useEffect(() => {
+    let observer = new IntersectionObserver(
+      (entries) => {
+        if (entries[0].isIntersecting) {
+          play();
+        } else {
+          pause();
+        }
+      },
+      {threshold: MIN_VISIBILITY_RATIO_FOR_AUTOPLAY}
+    );
+
+    observer.observe(wrapperRef.current);
+
+    return () => {
+      observer.disconnect();
+      observer = null;
+    };
+  }, [wrapperRef, play, pause]);
+
+  return (
+    <>
+      {displayIcon && (
+        <div
+          class={`amp-video-eq ${playing ? `amp-video-eq-play` : ''}`}
+          style={{display: 'flex'}}
+        >
+          {[1, 2, 3, 4].map((i) => (
+            <div class="amp-video-eq-col" key={i}>
+              <div class={`amp-video-eq-filler amp-video-eq-${i}-1`}></div>
+              <div class={`amp-video-eq-filler amp-video-eq-${i}-2`}></div>
+            </div>
+          ))}
+        </div>
+      )}
+
+      {displayOverlay && (
+        <div
+          style={{
+            position: 'absolute',
+            left: 0,
+            right: 0,
+            bottom: 0,
+            top: 0,
+            zIndex: 1,
+          }}
+          role="button"
+          onClick={onOverlayClick}
+        ></div>
+      )}
+
+      {/* TODO(wg-bento): Global styling.
+          https://github.com/ampproject/wg-bento/issues/7 */}
+      <style>{autoplayCss}</style>
```

Comment:
See here: https://github.com/ampproject/amphtml/pull/29551. Not merged yet. But heads up. And I might be able to merge it soon. Though it requires additional work for storybooks.

Reply:
For my own understanding: I'm assuming that this won't work on React mode?

Reply:
Correct. So not quite ready to switch yet. Just heads where that work is leaning toward. We should have more news about this very soon.